File: Sorting/Quicksort/Partition.java
package Sorting.Quicksort;

public class Partition {
    public static int partition(int[] A, int p, int r) {
        int x = A[r];
        int i = p - 1;

        for (int j = p; j < r; j++) {
            if (A[j] <= x) {
                i++;
                int temp = A[i];
                A[i] = A[j];
                A[j] = temp;
            }
        }
        
        int temp = A[i + 1];
        A[i + 1] = A[r];
        A[r] = temp;

        return i + 1;
    }
}

File: Sorting/Quicksort/Python/Partition.py
def Partition (A, p, r):
    x = A[r]
    i = p - 1

    for j in range (p, r):
        if A [j] <= x:
            i += 1
            A[i], A[j] = A[j], A[i]
    
    A[i + 1], A[r] = A[r], A[i + 1]
    return i + 1